py2hwsw/lib/hardware/arith_logic/iob_diff/iob_diff.py

```diff
@@ -83,17 +83,15 @@ def setup(py_params_dict):
                 "connect": {
                     "clk_en_rst_s": "clk_en_rst_s",
                     "data_i": "data_i",
-                    "data_o": "data_i_reg",
+                    "data_o": "data_int_reg",
                 },
             },
         ],
-        "snippets": [
-            {
-                "verilog_code": r"""
-                assign data_o = data_i - data_i_reg;
-                """
-            }
-        ],
+        "comb": {
+            "code": """
+                data_o = data_i - data_i_reg;
```

Comment:
data_o must have the is_var=True parameter, or the generated code won't work. The output will be set as a verilog `output wire`, when it must be a verilog `output reg`